### PR TITLE
Add structured logging on FXGraphCache hit

### PR DESCRIPTION
**Pull request by jamesjwu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #129588

We'll also want to do this for AOTAutogradCache once that's ready 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

Differential Revision: [D59144226](https://our.internmc.facebook.com/intern/diff/D59144226)

### CODE REVIEW

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -902,6 +902,10 @@ def iterate_over_candidates() -> Generator[CompiledFxGraph, None, None]:
             # Not expected, but in case the PyCodeCache entry is removed from
             # underneath us, treat it as a cache miss and recompile.
             log.error("Failed to load cached artifact: %s", artifact_path)
+            trace_structured(
```
Comment (oulgen) [inline torch/_inductor/codecache.py:905]: we dont need this, in this case we will compile as if no cache hit

File: torch/_inductor/codecache.py [hunk 2]
```diff
@@ -923,6 +927,12 @@ def iterate_over_candidates() -> Generator[CompiledFxGraph, None, None]:
 
         GraphLowering.save_output_code(code)
         output_code_log.debug("Output code: \n%s", code)
+        # On cache hit, use artifact path as filename
+        trace_structured(
+            "inductor_output_code",
+            lambda: {"filename": artifact_path},
```
Comment (oulgen) [inline torch/_inductor/codecache.py:933]: should we add something to indicate that this was a result of cache hit?

**Review by oulgen — APPROVED** [review]

File: torch/_inductor/codecache.py (hunk 2 above)
Comment (jamesjwu) [reply torch/_inductor/codecache.py:933]: I don't want to change the inductor_output_code interface just for FXGraphCache: I think it belongs in CompilationMetrics instead. We can work on some cache stats in compilation metrics to include both AOTAutogradCache and FXGraphCache, and log that directly via tlparse's compilation metrics parsing.

**Review by xmfan — APPROVED** [review]

**Comment by oulgen** [comment]
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59144226).

**Pull request closed without merging**